src/test/java/com/commercetools/sync/taxcategories/TaxCategorySyncOptionsTest.java

```diff
@@ -0,0 +1,158 @@
+        assertThat(taxCategorySyncOptions.getBeforeUpdateCallback()).isNotNull();
+
+        List<UpdateAction<TaxCategory>> updateActions = singletonList(SetKey.of("key"));
+        List<UpdateAction<TaxCategory>> filteredList = taxCategorySyncOptions
+            .applyBeforeUpdateCallBack(updateActions, mock(TaxCategoryDraft.class), mock(TaxCategory.class));
+
+        assertAll(
+            () -> assertThat(filteredList)
+                .as("returned 'updateActions' should not be equal to prepared ones")
+                .isNotEqualTo(updateActions),
+            () -> assertThat(filteredList).as("returned 'updateActions' should be empty").isEmpty()
+        );
+    }
+
+    private interface MockTriFunction extends
+        TriFunction<List<UpdateAction<TaxCategory>>, TaxCategoryDraft, TaxCategory, List<UpdateAction<TaxCategory>>> {
+    }
+
+    @Test
+    void applyBeforeUpdateCallback_WithEmptyUpdateActions_ShouldNotApplyBeforeUpdateCallback() {
+        TaxCategorySyncOptionsTest.MockTriFunction beforeUpdateCallback =
+            mock(TaxCategorySyncOptionsTest.MockTriFunction.class);
+
+        final TaxCategorySyncOptions taxCategorySyncOptions = TaxCategorySyncOptionsBuilder.of(CTP_CLIENT)
+            .beforeUpdateCallback(beforeUpdateCallback)
+            .build();
+
+        assertThat(taxCategorySyncOptions.getBeforeUpdateCallback()).isNotNull();
+
+        List<UpdateAction<TaxCategory>> updateActions = emptyList();
+        List<UpdateAction<TaxCategory>> filteredList = taxCategorySyncOptions
+            .applyBeforeUpdateCallBack(updateActions, mock(TaxCategoryDraft.class), mock(TaxCategory.class));
+
+        assertThat(filteredList).as("returned 'updateActions' should be empty").isEmpty();
+        verify(beforeUpdateCallback, never()).apply(any(), any(), any());
+    }
+
+    @Test
+    void applyBeforeUpdateCallback_WithCallback_ShouldReturnFilteredList() {
+        TriFunction<List<UpdateAction<TaxCategory>>, TaxCategoryDraft, TaxCategory, List<UpdateAction<TaxCategory>>>
+            beforeUpdateCallback = (updateActions, newCategory, oldCategory) -> emptyList();
+        final TaxCategorySyncOptions taxCategorySyncOptions = TaxCategorySyncOptionsBuilder.of(CTP_CLIENT)
+            .beforeUpdateCallback(beforeUpdateCallback)
+            .build();
+        assertThat(taxCategorySyncOptions.getBeforeUpdateCallback()).isNotNull();
+
+        List<UpdateAction<TaxCategory>> updateActions = singletonList(SetKey.of("key"));
+        List<UpdateAction<TaxCategory>> filteredList = taxCategorySyncOptions
+            .applyBeforeUpdateCallBack(updateActions, mock(TaxCategoryDraft.class), mock(TaxCategory.class));
+
+        assertAll(
+            () -> assertThat(filteredList)
+                .as("returned 'updateActions' should not be equal to prepared ones")
+                .isNotEqualTo(updateActions),
+            () -> assertThat(filteredList).as("returned 'updateActions' should be empty").isEmpty()
```

Comment:
I think the empty check here is enough, since we know for a fact that it is not equal to the updateActions variable which is not empty.